api/agent/iofs_notlinux.go

```diff
@@ -0,0 +1,21 @@
+// +build !linux
```

Comment:
it would be nice if this blew up at build time when a user tried to use this rather than waiting until runtime and blowing up when trying to run a container. this shouldn't even build to me, and agent would have to be configured with this, but yea i guess the bool may be easier to plumb, so as i'm getting used to saying, punt...

Reply:
@rdallman for it to blow up at build time it would have to be configured at build time not start time

Reply:
in any case, i think this use of build flags smells and isn't the proper way to use them... they should be used to block builds or have alternate implementations, not for runtime configuration errors... (at least let it be known...)

Reply:
yep. we can do it...

Reply:
we could leave it as something for 'extensions' to configure, when building their own fn binary with the agent and not using core's main (so not using env vars to configure)